source/common/router/config_impl.cc

```diff
@@ -482,6 +482,7 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
               ? route.route().host_rewrite_path_regex().substitution()
               : ""),
       append_xfh_(route.route().append_x_forwarded_host()), cluster_name_(route.route().cluster()),
+      path_stat_name_storage_(route.stat_prefix(), factory_context.scope().symbolTable()),
```

Comment:
Is there a way to initialize this only if stat prefix is there? Does it really matter?

Reply:
@jmarantz can help review this part. I would have to dig back in and relearn how all of this works at this point. I'm not sure if this is the right way or there is a better way.